components/econet/econet.cpp

```diff
@@ -225,8 +241,12 @@ void Econet::parse_message_(bool is_tx) {
             ESP_LOGE(TAG, "Unexpected length of %d for %s", item_len, datapoint_id.c_str());
             break;
           }
-          EconetDatapointType item_type = EconetDatapointType(pdata[tpos + 1] & 0x7F);
```

Comment:
Does this already set the enum value to UNSUPPORTED. From the linked issue my understanding is you get 0x15 here right? If that's the case you should be able to revert changes in this block of code.

Reply:
Fixed in https://github.com/esphome-econet/esphome-econet/pull/193